src/main/resources/schema/provider.definition.schema.v1.json

```
@@ -204,6 +204,14 @@
             "description": "A list of JSON pointers for properties that can only be updated under certain conditions. For example, you can upgrade the engine version of an RDS DBInstance but you cannot downgrade it.  When updating this property for a resource in a CloudFormation stack, the resource will be replaced if it cannot be updated.",
             "$ref": "base.definition.schema.v1.json#/definitions/jsonPointerArray"
         },
+        "nonPublicProperties": {
+            "description": "A list of JSON pointers for properties that are hidden. These properties will still be used but will not be visible",
```

I would probably make it more descriptive like the description for other properties.
```suggestion
            "description": "A list of JSON pointers for properties that are hidden. These properties can be used by developers to test new properties but will not be visible for customers to use",
```